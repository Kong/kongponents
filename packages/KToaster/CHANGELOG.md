# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 6.10.1 (2022-02-15)


### Bug Fixes

* **ktabs:** add bottom border for tabs ([#494](https://github.com/Kong/kongponents/issues/494)) ([5a6510c](https://github.com/Kong/kongponents/commit/5a6510cd902ef2c8e443f6b999e068bbee10278f))





# 6.10.0 (2022-02-15)


### Features

* **ktable:** add empty-state-action-button-icon ([#493](https://github.com/Kong/kongponents/issues/493)) ([d889728](https://github.com/Kong/kongponents/commit/d8897286a571c4fd6ac5abaa86e09c1e2e1b0b75))





## 6.9.7 (2022-02-15)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.9.6 (2022-02-11)


### Bug Fixes

* **composition-api:** fix usage of composition api ([#487](https://github.com/Kong/kongponents/issues/487)) ([c0827ca](https://github.com/Kong/kongponents/commit/c0827caec64bf055e1cc6a54b4f26ced5f811795))





## 6.9.5 (2022-02-10)


### Bug Fixes

* **styles:** remove tailwind from styles package ([#484](https://github.com/Kong/kongponents/issues/484)) ([1f1ee36](https://github.com/Kong/kongponents/commit/1f1ee36ea15b25f491b25872c01f6ba6f4a499c1))





## 6.9.4 (2022-02-07)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.9.3 (2022-02-05)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.9.2 (2022-02-04)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.9.1 (2022-02-04)

**Note:** Version bump only for package @kongponents/ktoaster





# 6.9.0 (2022-02-04)


### Features

* **kpop:** add positionFixed prop ([#481](https://github.com/Kong/kongponents/issues/481)) ([62dcbe6](https://github.com/Kong/kongponents/commit/62dcbe6cf6c82bf628316e3328f784cf32308eec))





## 6.8.1 (2022-02-03)

**Note:** Version bump only for package @kongponents/ktoaster





# 6.8.0 (2022-02-03)


### Features

* **ktable:** add legacy ktable component ([#477](https://github.com/Kong/kongponents/issues/477)) ([4454617](https://github.com/Kong/kongponents/commit/445461705d25b27426e2b542e65e8ad9a351a921))





## 6.7.11 (2022-01-21)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.7.10 (2022-01-12)


### Bug Fixes

* **ktable:** update sort styling ([#473](https://github.com/Kong/kongponents/issues/473)) ([d2d242f](https://github.com/Kong/kongponents/commit/d2d242fbeaf92db112f699cbe8eed163b066feff))





## 6.7.9 (2022-01-07)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.7.8 (2022-01-07)


### Bug Fixes

* **kinput:** allow numbers to be passed in as value ([#470](https://github.com/Kong/kongponents/issues/470)) ([fbf4c00](https://github.com/Kong/kongponents/commit/fbf4c00291383707770076110dcfa9519d31faef))





## 6.7.7 (2021-12-21)


### Reverts

* **version:** revert independent version fix ([#469](https://github.com/Kong/kongponents/issues/469)) ([b647e56](https://github.com/Kong/kongponents/commit/b647e56b5c05c25e55362a845e1c4a30fe40d910))





## 6.7.6 (2021-12-21)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.7.3 (2021-12-20)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.7.2 (2021-12-17)


### Bug Fixes

* **actions:** fix actions ([#459](https://github.com/Kong/kongponents/issues/459)) ([9bb5ad5](https://github.com/Kong/kongponents/commit/9bb5ad57272f9f57c140646eb9d4413d595115d6))





## 6.7.1 (2021-12-17)

**Note:** Version bump only for package @kongponents/ktoaster





# 6.7.0 (2021-12-17)


### Features

* **kinput:** add css variable for optional kinput placeholder text color overrides ([#443](https://github.com/Kong/kongponents/issues/443)) ([3b40207](https://github.com/Kong/kongponents/commit/3b40207eabd87b21568ec2697ad39961c2a91b40))





## 6.6.5 (2021-12-17)


### Bug Fixes

* **kpagination:** fix currentPage prop ([#455](https://github.com/Kong/kongponents/issues/455)) ([5e1a801](https://github.com/Kong/kongponents/commit/5e1a801114d2357a2d49dca3bb2571dd055b4119))





## 6.6.4 (2021-12-16)


### Bug Fixes

* **ktextarea:** fix v-model ([#454](https://github.com/Kong/kongponents/issues/454)) ([35f581f](https://github.com/Kong/kongponents/commit/35f581f5c1be13fdebf1a63b8742181de8e19a8b))





## 6.6.3 (2021-12-15)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.6.2 (2021-12-10)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.6.1 (2021-12-10)


### Bug Fixes

* **kalert:** fix invalid html structure ([#452](https://github.com/Kong/kongponents/issues/452)) ([15eade8](https://github.com/Kong/kongponents/commit/15eade87d0661ae7716a6e32f1fa63279dbf8743))





# 6.6.0 (2021-12-08)


### Features

* **comment:** add global comment in output file ([#451](https://github.com/Kong/kongponents/issues/451)) ([b2f6154](https://github.com/Kong/kongponents/commit/b2f6154c37be30a3d364337f3d87fe40a64fc753))





## 6.5.9 (2021-12-01)


### Bug Fixes

* infinite publish ([d3f275b](https://github.com/Kong/kongponents/commit/d3f275bb668dd95782f6150232cc22a5b95b1bcf))





## [6.5.8](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@6.5.7...@kongponents/ktoaster@6.5.8) (2021-12-01)

**Note:** Version bump only for package @kongponents/ktoaster





## [6.5.7](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@6.5.6...@kongponents/ktoaster@6.5.7) (2021-12-01)

**Note:** Version bump only for package @kongponents/ktoaster





## [6.5.6](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@6.5.5...@kongponents/ktoaster@6.5.6) (2021-12-01)

**Note:** Version bump only for package @kongponents/ktoaster





## [6.5.5](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@6.5.4...@kongponents/ktoaster@6.5.5) (2021-12-01)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.5.4 (2021-12-01)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.5.3 (2021-11-30)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.5.2 (2021-11-29)


### Bug Fixes

* **ktable:** allow init client sort ([c3e2a69](https://github.com/Kong/kongponents/commit/c3e2a693974d6aae7f829841af89391c5dd357b2))





## 6.5.1 (2021-11-24)


### Bug Fixes

* **ktable/kcardcatalog:** fetch after init ([#442](https://github.com/Kong/kongponents/issues/442)) ([4a053ae](https://github.com/Kong/kongponents/commit/4a053ae4e39e7df7acc798510c8a2fa9e5ab3f6c))





# 6.5.0 (2021-11-24)


### Features

* **ktable:** add cacheKey prop ([#441](https://github.com/Kong/kongponents/issues/441)) ([743082d](https://github.com/Kong/kongponents/commit/743082df383699b8016d3b6d85497ec77563aaec))





# 6.4.0 (2021-11-24)


### Features

* **ktable:** default sort to asc ([#440](https://github.com/Kong/kongponents/issues/440)) ([95f2e91](https://github.com/Kong/kongponents/commit/95f2e9164c09e50b25f2e334ff30231ab0984503))





# 6.3.0 (2021-11-23)


### Features

* **ktable/kcardcatalog:** make states slottable ([#438](https://github.com/Kong/kongponents/issues/438)) ([43eac93](https://github.com/Kong/kongponents/commit/43eac9337193fe47b28b27576d727e6cbbabad50))





# 6.2.0 (2021-11-23)


### Features

* **ktable/kcardcatalog:** emit cta click events ([#437](https://github.com/Kong/kongponents/issues/437)) ([29eb2a5](https://github.com/Kong/kongponents/commit/29eb2a53cf180935afee1ee10f1b88b462dc29ad))





## 6.1.19 (2021-11-23)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.1.18 (2021-11-23)


### Bug Fixes

* **ktable:** add missing style ([#435](https://github.com/Kong/kongponents/issues/435)) ([8782dee](https://github.com/Kong/kongponents/commit/8782dee82dafc68c1f956149a249b950359401b1))





## 6.1.17 (2021-11-22)


### Bug Fixes

* fix deps and build process ([#434](https://github.com/Kong/kongponents/issues/434)) ([9fdd180](https://github.com/Kong/kongponents/commit/9fdd180c9816bd4a43d4cd9229fc45b180ba5eed))





## 6.1.16 (2021-11-19)


### Bug Fixes

* **publish:** update node version in publish ([#433](https://github.com/Kong/kongponents/issues/433)) ([26cfe34](https://github.com/Kong/kongponents/commit/26cfe34b6fd9fbb3d61ffdb95412804bac88ba36))





## 6.1.15 (2021-11-19)


### Bug Fixes

* **build:** update lerna ignore flag ([#431](https://github.com/Kong/kongponents/issues/431)) ([768c5e4](https://github.com/Kong/kongponents/commit/768c5e4e78bfca118fe7b2d022778854e076b323))





## 6.1.14 (2021-11-19)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.1.13 (2021-11-19)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.1.12 (2021-11-19)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.1.11 (2021-11-19)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.1.10 (2021-11-18)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.1.9 (2021-11-18)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.1.8 (2021-11-18)


### Bug Fixes

* **css:** box-shadow does not render on ios ([#420](https://github.com/Kong/kongponents/issues/420)) ([c0ba7cb](https://github.com/Kong/kongponents/commit/c0ba7cb90bf3c158a1baa6b879d33230359db830))





## 6.1.7 (2021-11-17)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.1.6 (2021-11-17)


### Bug Fixes

* **ktextarea:** limit exceeded event ([#418](https://github.com/Kong/kongponents/issues/418)) ([f7cf421](https://github.com/Kong/kongponents/commit/f7cf421a4ded7f5b99eaa5a18d02a27d450020b7))





## 6.1.5 (2021-11-17)

**Note:** Version bump only for package @kongponents/ktoaster





## 6.1.4 (2021-11-17)


### Bug Fixes

* general fixes ([#415](https://github.com/Kong/kongponents/issues/415)) ([5b4bb48](https://github.com/Kong/kongponents/commit/5b4bb484d74e75779d39cf901b2cdaa23818e8e7))





## 6.1.3 (2021-11-17)

**Note:** Version bump only for package @kongponents/ktoaster





## [6.1.2](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@5.0.8...@kongponents/ktoaster@6.1.2) (2021-11-17)

Kongponents redesign - no changes to this component


## [5.0.8](https://github.com/Kong/kongponents/compare/v5.0.7...v5.0.8) (2021-11-15)

**Note:** Version bump only for package @kongponents/ktoaster





## [5.0.7](https://github.com/Kong/kongponents/compare/v5.0.6...v5.0.7) (2021-10-26)

**Note:** Version bump only for package @kongponents/ktoaster





## [5.0.6](https://github.com/Kong/kongponents/compare/v5.0.5...v5.0.6) (2021-10-26)

**Note:** Version bump only for package @kongponents/ktoaster





## [5.0.5](https://github.com/Kong/kongponents/compare/v5.0.4...v5.0.5) (2021-10-13)

**Note:** Version bump only for package @kongponents/ktoaster





## [5.0.4](https://github.com/Kong/kongponents/compare/v5.0.3...v5.0.4) (2021-10-06)

**Note:** Version bump only for package @kongponents/ktoaster





## [5.0.3](https://github.com/Kong/kongponents/compare/v5.0.2...v5.0.3) (2021-09-22)

**Note:** Version bump only for package @kongponents/ktoaster





## [5.0.2](https://github.com/Kong/kongponents/compare/v5.0.1...v5.0.2) (2021-09-15)

**Note:** Version bump only for package @kongponents/ktoaster





## [5.0.1](https://github.com/Kong/kongponents/compare/v5.0.0...v5.0.1) (2021-09-13)

**Note:** Version bump only for package @kongponents/ktoaster





# [5.0.0](https://github.com/Kong/kongponents/compare/v4.0.2...v5.0.0) (2021-09-13)

**Note:** Version bump only for package @kongponents/ktoaster





## [4.0.2](https://github.com/Kong/kongponents/compare/v4.0.1...v4.0.2) (2021-09-09)

**Note:** Version bump only for package @kongponents/ktoaster





## [4.0.1](https://github.com/Kong/kongponents/compare/v4.0.0...v4.0.1) (2021-09-07)

**Note:** Version bump only for package @kongponents/ktoaster





# 4.0.0 (2021-09-03)


### Bug Fixes

* **cleanup:** update versions & add scoped to KModal ([#19](https://github.com/Kong/kongponents/issues/19)) ([a16e216](https://github.com/Kong/kongponents/commit/a16e216c27feebce1599c2f68590390f679bbf15))
* **dist:** dist to all packages ([#152](https://github.com/Kong/kongponents/issues/152)) ([217c46b](https://github.com/Kong/kongponents/commit/217c46b02afe960f8fa597ba77410dcf034b1716))
* **KToaster:** remove left margin on close button ([#293](https://github.com/Kong/kongponents/issues/293)) ([45d9be5](https://github.com/Kong/kongponents/commit/45d9be5007dbe4f000dd35b794d03f4d6fe724ba))
* **scoped styles:** updated scoped styles type ([#18](https://github.com/Kong/kongponents/issues/18)) ([ba26b73](https://github.com/Kong/kongponents/commit/ba26b733387ab0f8816c46a127d0fc9532cf7395))
* **taoster:** add correct default values to toaster ([#12](https://github.com/Kong/kongponents/issues/12)) ([eba5bd2](https://github.com/Kong/kongponents/commit/eba5bd26fb96df87a5c0d74e9d6cd68181784fec))


### Features

* 🍞 Add KToaster for all our avocado toast needs ([#7](https://github.com/Kong/kongponents/issues/7)) ([660a1e4](https://github.com/Kong/kongponents/commit/660a1e43d4d3a154051021dc146ea707efd56443))
* **KIcon:** add svgElements slot ([#302](https://github.com/Kong/kongponents/issues/302)) ([17fc9c5](https://github.com/Kong/kongponents/commit/17fc9c537027a301780e5946e43e3c0eaeb6a159))
* **ktoaster:** add ToastManager API ([#140](https://github.com/Kong/kongponents/issues/140)) ([63d0a46](https://github.com/Kong/kongponents/commit/63d0a46128d51bc7eb55cd9e5793a4eec76a1322)), closes [#156](https://github.com/Kong/kongponents/issues/156)
* **KToaster:** change to white style, move to bottom ([#243](https://github.com/Kong/kongponents/issues/243)) ([70250af](https://github.com/Kong/kongponents/commit/70250afb513817c861cc4fbe93026f1a45a32fdb))


### BREAKING CHANGES

* **ktoaster:** ktoaster now imported differently
and does not have since KToaster.vue anymore, as now
it is managed by ToastManager

* chore(KToaster) update license to apache 2.0





## [1.2.5](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.2.4...@kongponents/ktoaster@1.2.5) (2021-07-27)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.2.4](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.2.3...@kongponents/ktoaster@1.2.4) (2021-07-12)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.2.3](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.2.2...@kongponents/ktoaster@1.2.3) (2021-06-29)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.2.2](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.2.1...@kongponents/ktoaster@1.2.2) (2021-06-08)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.2.1](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.2.0...@kongponents/ktoaster@1.2.1) (2021-05-17)

**Note:** Version bump only for package @kongponents/ktoaster





# [1.2.0](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.1.12...@kongponents/ktoaster@1.2.0) (2021-05-11)


### Features

* **KIcon:** add svgElements slot ([#302](https://github.com/Kong/kongponents/issues/302)) ([17fc9c5](https://github.com/Kong/kongponents/commit/17fc9c537027a301780e5946e43e3c0eaeb6a159))





## [1.1.12](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.1.11...@kongponents/ktoaster@1.1.12) (2021-04-12)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.1.11](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.1.10...@kongponents/ktoaster@1.1.11) (2021-03-31)


### Bug Fixes

* **KToaster:** remove left margin on close button ([#293](https://github.com/Kong/kongponents/issues/293)) ([45d9be5](https://github.com/Kong/kongponents/commit/45d9be5007dbe4f000dd35b794d03f4d6fe724ba))





## [1.1.10](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.1.9...@kongponents/ktoaster@1.1.10) (2021-03-19)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.1.9](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.1.8...@kongponents/ktoaster@1.1.9) (2021-01-08)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.1.8](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.1.7...@kongponents/ktoaster@1.1.8) (2020-11-09)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.1.7](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.1.6...@kongponents/ktoaster@1.1.7) (2020-10-09)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.1.6](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.1.5...@kongponents/ktoaster@1.1.6) (2020-09-22)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.1.5](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.1.4...@kongponents/ktoaster@1.1.5) (2020-09-03)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.1.4](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.1.3...@kongponents/ktoaster@1.1.4) (2020-09-03)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.1.3](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.1.2...@kongponents/ktoaster@1.1.3) (2020-09-01)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.1.2](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.1.1...@kongponents/ktoaster@1.1.2) (2020-08-25)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.1.1](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.1.0...@kongponents/ktoaster@1.1.1) (2020-08-10)

**Note:** Version bump only for package @kongponents/ktoaster





# [1.1.0](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.0.18...@kongponents/ktoaster@1.1.0) (2020-07-24)


### Features

* **KToaster:** change to white style, move to bottom ([#243](https://github.com/Kong/kongponents/issues/243)) ([70250af](https://github.com/Kong/kongponents/commit/70250afb513817c861cc4fbe93026f1a45a32fdb))





## [1.0.18](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.0.17...@kongponents/ktoaster@1.0.18) (2020-07-23)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.0.17](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.0.16...@kongponents/ktoaster@1.0.17) (2020-07-08)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.0.16](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.0.15...@kongponents/ktoaster@1.0.16) (2020-07-06)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.0.15](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.0.14...@kongponents/ktoaster@1.0.15) (2020-07-01)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.0.14](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.0.13...@kongponents/ktoaster@1.0.14) (2020-06-30)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.0.13](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.0.12...@kongponents/ktoaster@1.0.13) (2020-06-29)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.0.12](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.0.11...@kongponents/ktoaster@1.0.12) (2020-06-25)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.0.11](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.0.10...@kongponents/ktoaster@1.0.11) (2020-03-25)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.0.10](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.0.9...@kongponents/ktoaster@1.0.10) (2020-03-19)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.0.9](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.0.8...@kongponents/ktoaster@1.0.9) (2020-03-11)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.0.8](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.0.7...@kongponents/ktoaster@1.0.8) (2020-03-06)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.0.7](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.0.6...@kongponents/ktoaster@1.0.7) (2020-02-28)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.0.6](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.0.4...@kongponents/ktoaster@1.0.6) (2020-01-16)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.0.4](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.0.3...@kongponents/ktoaster@1.0.4) (2020-01-09)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.0.3](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.0.2...@kongponents/ktoaster@1.0.3) (2019-12-11)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.0.2](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.0.1...@kongponents/ktoaster@1.0.2) (2019-11-12)

**Note:** Version bump only for package @kongponents/ktoaster





## [1.0.1](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@1.0.0...@kongponents/ktoaster@1.0.1) (2019-11-12)

**Note:** Version bump only for package @kongponents/ktoaster





# [1.0.0](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@0.1.5...@kongponents/ktoaster@1.0.0) (2019-11-12)


### Features

* **ktoaster:** add ToastManager API ([#140](https://github.com/Kong/kongponents/issues/140)) ([63d0a46](https://github.com/Kong/kongponents/commit/63d0a46128d51bc7eb55cd9e5793a4eec76a1322)), closes [#156](https://github.com/Kong/kongponents/issues/156)


### BREAKING CHANGES

* **ktoaster:** ktoaster now imported differently
and does not have since KToaster.vue anymore, as now
it is managed by ToastManager

* chore(KToaster) update license to apache 2.0





# [2.0.0](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@0.1.5...@kongponents/ktoaster@2.0.0) (2019-11-12)


### Features

* **ktoaster:** add ToastManager API ([#140](https://github.com/Kong/kongponents/issues/140)) ([63d0a46](https://github.com/Kong/kongponents/commit/63d0a46128d51bc7eb55cd9e5793a4eec76a1322)), closes [#156](https://github.com/Kong/kongponents/issues/156)


### BREAKING CHANGES

* **ktoaster:** ktoaster now imported differently
and does not have since KToaster.vue anymore, as now
it is managed by ToastManager

* chore(KToaster) update license to apache 2.0





# [1.0.0](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@0.1.5...@kongponents/ktoaster@1.0.0) (2019-11-12)


### Features

* **ktoaster:** add ToastManager API ([#140](https://github.com/Kong/kongponents/issues/140)) ([63d0a46](https://github.com/Kong/kongponents/commit/63d0a46128d51bc7eb55cd9e5793a4eec76a1322)), closes [#156](https://github.com/Kong/kongponents/issues/156)


### BREAKING CHANGES

* **ktoaster:** ktoaster now imported differently
and does not have since KToaster.vue anymore, as now
it is managed by ToastManager

* chore(KToaster) update license to apache 2.0





## [0.1.5](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@0.1.4...@kongponents/ktoaster@0.1.5) (2019-11-11)

**Note:** Version bump only for package @kongponents/ktoaster





## [0.1.4](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@0.1.3...@kongponents/ktoaster@0.1.4) (2019-11-08)


### Bug Fixes

* **dist:** dist to all packages ([#152](https://github.com/Kong/kongponents/issues/152)) ([217c46b](https://github.com/Kong/kongponents/commit/217c46b02afe960f8fa597ba77410dcf034b1716))





## [0.1.3](https://github.com/Kong/kongponents/compare/@kongponents/ktoaster@0.1.2...@kongponents/ktoaster@0.1.3) (2019-11-08)

**Note:** Version bump only for package @kongponents/ktoaster
