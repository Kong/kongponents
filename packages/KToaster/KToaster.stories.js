import Vue from 'vue'
import { storiesOf } from '@storybook/vue'
import VueInfoAddon from 'storybook-addon-vue-info'

storiesOf('KToaster', module)
  .addDecorator(VueInfoAddon)
  .add('Default', () => ({
    
  }))
