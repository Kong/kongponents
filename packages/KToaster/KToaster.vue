<template>
  <transition name="slide">
    <div
      v-show="isVisible"
      :class="{visible: isVisible}"
      class="k-toaster">
      <div
        :class="appearance"
        class="k-toast">
        <a
          class="k-toast-btn-clear"
          @click="close()"/>
        <!-- @slot to add custom message if not passing :message prop -->
        <slot :name="message">{{ message }}</slot>
      </div>
    </div>
  </transition>
</template>

<script>
export default {
  name: 'KToaster',
  props: {
    /**
      *  Pass whether or not the modal should be visible
      */
    isVisible: {
      type: Boolean,
      required: true
    },
    /**
      * Message to show in toaster
      */
    message: {
      type: String,
      required: true
    },
    /**
      * Amount of time in milliseconds before toaster hides
      */
    timeoutMilliseconds: {
      type: Number,
      default: 10000
    },
    /**
      * Color variantion of Toaster.<br>
      * Options: success, danger, info or warning
      */
    appearance: {
      type: String,
      default: 'info',
      validator: function (value) {
        return [
          'danger',
          'success',
          'info',
          'warning'
        ].indexOf(value) !== -1
      },
      required: false
    }
  },

  data () {
    return {
      toasterTimeout: null
    }
  },

  watch: {
    isVisible (newValue, oldValue) {
      if (newValue === true && newValue !== oldValue) {
        this.toasterTimeout = setTimeout(() => this.close(), this.timeoutMilliseconds)
      }
    }
  },

  methods: {
    close () {
      clearTimeout(this.toasterTimeout)
      this.$emit('close')
    }
  }
}
</script>

<style scoped>
.k-toaster {
  position: fixed;
  top: 50px;
  right: 0;
  z-index: 10000;
  width: 300px;
  padding-left: 10px;
  padding-right: 10px;
}
.k-toaster .k-toast {
  display: flex;
  align-items: center;
  margin-bottom: 10px;
  padding: 1rem;
  border: 1px solid;
  border-radius: 8px;
  color: #383d41;
  background-color: #e2e3e5;
  border-color: #d6d8db;
  transition: all 0.3s ease;
}
.k-toaster .k-toast .k-toast-btn-clear {
  order: 1;
  margin-left: auto;
  border: 0;
  color: inherit;
  opacity: 0.45;
  text-decoration: none;
  cursor: pointer;
  background: transparent;
}
.k-toaster .k-toast .k-toast-btn-clear:hover {
  opacity: 0.85;
}
.k-toaster .k-toast .k-toast-btn-clear::before {
  content: "\2715";
}

/* Toaster Variants */
.k-toaster .k-toast.success {
  color: #008038;
  border-color: #8cd9ac;
  background: #ecf9f2;
}
.k-toaster .k-toast.info {
  color: #004a80;
  border-color: #7cc9ff;
  background: #ebf7ff;
}
.k-toaster .k-toast.warning {
  color: #403624;
  border-color: #FFDF80;
  background-color: #FFF7E8;
}
.k-toaster .k-toast.danger {
  color: #c20d0a;
  border-color: #ff8080;
  background: #fff0f0;
}

/* Toaster Slide Animation */
.slide-enter-active,
.slide-leave-active {
  transition: all .4s ease;
}
.slide-enter,
.slide-leave-to {
  transform: translate(100%);
}

@media (max-width: 300px) {
  .k-toaster {
    width: 100%;
  }
}
</style>
