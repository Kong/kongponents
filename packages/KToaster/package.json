{
  "name": "@kongponents/ktoaster",
  "version": "0.1.2",
  "description": "Toaster component",
  "main": "KToaster.vue",
  "scripts": {
    "test": "echo \"Error: no test specified\""
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Kong/kongponents.git"
  },
  "author": "Kong Inc",
  "license": "ISC",
  "bugs": {
    "url": "https://github.com/Kong/kongponents/issues"
  },
  "homepage": "https://github.com/Kong/kongponents#readme"
}
