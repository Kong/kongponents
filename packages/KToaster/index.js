import KToaster from './KToaster.vue'
import ToastManager from './ToastManager.js'

const plugin = {
  install (Vue) {
    Vue.component('KToaster', KToaster)
  }
}

export default plugin
export { KToaster, ToastManager }

if (typeof window !== 'undefined' && window.Vue) {
  window.Vue.use(plugin)
}
