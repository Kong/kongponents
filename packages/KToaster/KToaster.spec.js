import { mount, config } from '@vue/test-utils'
import KToaster from '@/KToaster/KToaster'
config.silent = true

const variants = ['success', 'danger', 'warning', 'info']

let rendersCorrectVariant = (variant) => {
  it(`renders ${variant} variant`, () => {
    const wrapper = mount(KToaster, {
      propsData: {
        'appearance': `${variant}`,
        'isVisible': true,
        'message': 'cool test'
      }
    })

    expect(wrapper.find('.k-toast').classes()).toContain(`${variant}`)
    expect(wrapper.html()).toMatchSnapshot()
  })
}

describe('KToaster', () => {
  variants.map(v => rendersCorrectVariant(v))

  it('renders info variant when no appearance prop', () => {
    const wrapper = mount(KToaster, {
      propsData: {
        'message': 'no variant',
        'isVisible': true
      }
    })

    expect(wrapper.find('.k-toast').classes()).toContain('info')
  })
})
