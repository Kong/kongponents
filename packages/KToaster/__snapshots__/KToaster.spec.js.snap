// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KToaster renders danger variant 1`] = `
<div class="k-toaster visible" name="slide">
  <div class="k-toast danger"><a class="k-toast-btn-clear"></a> cool test</div>
</div>
`;

exports[`KToaster renders info variant 1`] = `
<div class="k-toaster visible" name="slide">
  <div class="k-toast info"><a class="k-toast-btn-clear"></a> cool test</div>
</div>
`;

exports[`KToaster renders success variant 1`] = `
<div class="k-toaster visible" name="slide">
  <div class="k-toast success"><a class="k-toast-btn-clear"></a> cool test</div>
</div>
`;

exports[`KToaster renders warning variant 1`] = `
<div class="k-toaster visible" name="slide">
  <div class="k-toast warning"><a class="k-toast-btn-clear"></a> cool test</div>
</div>
`;
